api/python/t4/snapshots.py

```diff
@@ -637,7 +637,7 @@ def _materialize(self, path):
                 new_physical_key = path + "/" + logical_key
                 new_type = PhysicalKeyType.S3.name
             else:
-                new_physical_key = os.path.join(path, logical_key)
+                new_physical_key = os.path.join(path, logical_key).resolve().as_uri()
```

Comment:
This won't work, though: `os.path.join` gives you a string, not a Path. In any case... you should have exactly the same logic in `if` and in `else`, since it's a URI in both cases.